### DISCUSSION TITLE
Switching from sqlite to postgres

### DISCUSSION THREAD

**Discussion by Jimmy99** [benbjohnson/wtf#53] [Q&A] [answered]
What would be the correct approach to switch the storage from sqlite to postgres?
I guess adding a subpackage called postgres with the relevant code to connect to a postgres instance, but I am not sure how the other services use the sqlite storage and how to switch that to postgres

**Comment by benbjohnson** [comment, marked as answer]
Hi @Jimmy99. Good question. It depends if you need to support both SQLite & Postgres or if you're just moving to Postgres.

 If you're moving to Postgres (and removing SQLite support) then I would create a `postgres` package for the code and update the `sqlite` references in  `cmd/wtfd/main.go`. The other packages shouldn't have knowledge of the underlying implementation—only the `main` package that wires it all together.

If you're supporting both Postgres & SQLite, you could either add a `postgres` package and keep the `sqlite` package or you could look at implementing SQL support using some kind of generic ORM tool. In that case, I would move the code into a `sql` package that would support both.
